File: server/apps/lessons/application/dto/course_lesson_plan_dto.py
from dataclasses import dataclass


@dataclass
class CourseLessonPlanDTO:
    course_lesson_plan_id: str

    university_id: str
    university_name: str

    subject_id: str
    subject_name: str

    semester_id: str
    semester_name: str
    semester_number: str

    academic_year_id: str
    academic_year_name: str

    total_lessons: int

File: client/src/application/lessons/dto/course_lesson_plan.dto.js
class CourseLessonPlanDTO {
  constructor(data = {}) {
    this.courseLessonPlanId =
      data.course_lesson_plan_id ?? data.courseLessonPlanId;

    this.universityId = data.university_id ?? data.universityId;

    this.universityName = data.university_name ?? data.universityName;

    this.subjectId = data.subject_id ?? data.subjectId;

    this.subjectName = data.subject_name ?? data.subjectName;

    this.semesterId = data.semester_id ?? data.semesterId;

    this.semesterName = data.semester_name ?? data.semesterName;

    this.semesterNumber = data.semester_number ?? data.semesterNumber;

    this.academicYearId = data.academic_year_id ?? data.academicYearId;

    this.academicYearName = data.academic_year_name ?? data.academicYearName;

    this.totalLessons = data.total_lessons ?? data.totalLessons;
  }
}

module.exports = CourseLessonPlanDTO;
